larpandoracontent/LArControlFlow/NeutrinoIdTool.cc

```diff
@@ -231,59 +249,101 @@ void NeutrinoIdTool<T>::SelectAllPfos(const pandora::Algorithm *const pAlgorithm
 }
 
 //------------------------------------------------------------------------------------------------------------------------------------------
-
 template<typename T>
-void NeutrinoIdTool<T>::SelectPfosByProbability(const pandora::Algorithm *const pAlgorithm, const SliceHypotheses &nuSliceHypotheses, const SliceHypotheses &crSliceHypotheses, const SliceFeaturesVector &sliceFeaturesVector, PfoList &selectedPfos) const
-{
+void NeutrinoIdTool<T>::SelectPfosByProbability(const pandora::Algorithm *const pAlgorithm, const SliceHypotheses &nuSliceHypotheses, const SliceHypotheses &crSliceHypotheses, const SliceFeaturesVector &sliceFeaturesVector, PfoList &selectedPfos, const PfoToFloatMap &pfoToProbabilityMap) const
+  {
+
     // Calculate the probability of each slice that passes the minimum probability cut
     std::vector<UintFloatPair> sliceIndexProbabilityPairs;
     for (unsigned int sliceIndex = 0, nSlices = nuSliceHypotheses.size(); sliceIndex < nSlices; ++sliceIndex)
-    {
-        const float nuProbability(sliceFeaturesVector.at(sliceIndex).GetNeutrinoProbability(m_mva));
+      {
+	std::vector<float> downProbCr;
+	std::vector<float> downProbNu;
+  
 
         for (const ParticleFlowObject *const pPfo : crSliceHypotheses.at(sliceIndex))
-        {
-            object_creation::ParticleFlowObject::Metadata metadata;
-            metadata.m_propertiesToAdd["NuScore"] = nuProbability;
-            PANDORA_THROW_RESULT_IF(STATUS_CODE_SUCCESS, !=, PandoraContentApi::ParticleFlowObject::AlterMetadata(*pAlgorithm, pPfo, metadata));
-        }
+	  {
+	    object_creation::ParticleFlowObject::Metadata metadata;
+	    
+	    auto search = pfoToProbabilityMap.find(pPfo);
+	    if (search !=  pfoToProbabilityMap.end()) {
+	      metadata.m_propertiesToAdd["downProb"] = search->second;
+	      if(search->second != -1) {
+		downProbCr.push_back(search->second);
+	      }
+	    }
+	  }
 
         for (const ParticleFlowObject *const pPfo : nuSliceHypotheses.at(sliceIndex))
-        {
-            object_creation::ParticleFlowObject::Metadata metadata;
+	  {
+	    object_creation::ParticleFlowObject::Metadata metadata;
+	    CaloHitList collectedHits;
+	    LArPfoHelper::GetCaloHits(pPfo, TPC_VIEW_U, collectedHits);
+	    LArPfoHelper::GetCaloHits(pPfo, TPC_VIEW_V, collectedHits);
+	    LArPfoHelper::GetCaloHits(pPfo, TPC_VIEW_W, collectedHits);
+
+	    PfoList daughterPfos = pPfo->GetDaughterPfoList();
+	    
+	    for (const ParticleFlowObject *const pPPfo : daughterPfos) {
```

Comment:
Can this all be collected up into a new method, which is cleanly called (in just one/two lines) in the existing implementation?

Reply:
It can; done.